### DISCUSSION TITLE
Are there too many OpCall here?

### DISCUSSION THREAD

**Discussion by zhuliquan** [antonmedv/expr#615] [General]
I find there are many type of OpCall OpCode in below code.
https://github.com/expr-lang/expr/blob/7772ea0fee5c9b153439a0ad5ee2e2b1a764f453/vm/opcodes.go#L55-L59
I read the code of `cmpiler.go` as below:
https://github.com/expr-lang/expr/blob/7772ea0fee5c9b153439a0ad5ee2e2b1a764f453/compiler/compiler.go#L157-L171
 I got that you want to generate special bytecode based on the number of arguments of the function. However, `OpCallN` is also can cover the `OpCall0` to the `OpCall3`. Now, is it a reasonable choice to delete those OpCode (i.e. `OpCall0` to `OpCall3`) now?

**Comment by antonmedv** [comment]
All of those opcodes are needed. OpCall1-3 is a special fast call type. OpCallN is a generic fallback.
